subprojects/core/src/main/java/org/gradle/api/internal/AbstractTask.java

```diff
@@ -559,7 +560,7 @@ public Task doFirst(final Closure action) {
         }
         taskMutator.mutate("Task.doFirst(Closure)", new Runnable() {
             public void run() {
-                getTaskActions().add(0, convertClosureToAction(action));
+                getTaskActions().add(0, convertClosureToAction(action, "doFirst {}"));
```

Comment:
We should do this for `doFirst(Action)` as well (unless the `Action` is already describable). Same for `doLast(Action)`.